File: yilu_an_frontend/miniprogram/api/binding.ts
// 绑定相关接口
import { api } from '../utils/request';

// 创建绑定关系请求参数
export interface CreateBindingParams {
  elderly_phone: string;
}
// 对应后端新增的 BindingUser 子资源
export interface BindingUser {
  user_id: number;
  phone: string;
  nickname: string | null;
  avatar_url: string | null;
}
// 绑定关系资源表述 (完全对标后端的 BindingResponse)
export interface Binding {
  binding_id: number;
  status: 'pending' | 'accepted' | 'rejected'; // 使用联合类型确保安全
  created_at: string;
  approved_at: string | null;
  elderly: BindingUser | null; // 嵌套对象
  family: BindingUser | null;  // 嵌套对象
}
// 解除绑定关系请求参数
export interface UnbindParams {
  elderly_phone: string;
}

// 绑定关系
// export interface Binding {
//   binding_id: number;
//   elderly_id: number;
//   elderly_nickname: string;
//   elderly_phone: string | null;
//   family_id: number;
//   family_nickname: string;
//   family_phone: string | null;
//   status: string;
//   created_at: string;
//   approved_at: string | null;
// }

// 绑定相关API
export const bindingApi = {
  // 创建绑定关系
  createBinding: (data: CreateBindingParams) => {
    return api.post<Binding>('/bindings/', data);
  },

  // 获取绑定关系列表
  getBindings: () => {
    return api.get<Binding[]>('/bindings/');
  },

  // 解除绑定关系
  // unbind: (data: UnbindParams) => {
  //   return api.delete('/bindings', {data});
  // },
// 解除绑定关系 (遵循 REST：DELETE 请求不带 Body，通过 URL 路径传 ID)
deleteBinding: (bindingId: number) => {
  return api.delete(`/bindings/${bindingId}`);
},
//   // 批准绑定请求
//   approveBinding: (bindingId: number) => {
//     return api.put<Binding>(`/bindings/${bindingId}/approve`);
//   },

//   // 拒绝绑定请求
//   rejectBinding: (bindingId: number) => {
//     return api.put<Binding>(`/bindings/${bindingId}/reject`);
//   },
  // 批准绑定请求
  approveBinding: (bindingId: number) => {
    return api.put<Binding>(`/bindings/${bindingId}`, {
      status: "accepted" 
    });
  },

  // 拒绝绑定请求
  rejectBinding: (bindingId: number) => {
    return api.put<Binding>(`/bindings/${bindingId}`, {
      status: "rejected"
    });
  },
};

File: yilu_an_backend/app/schemas/binding.py
from pydantic import BaseModel,Field,ConfigDict
from typing import Optional
from datetime import datetime
from enum import Enum

class BindingStatus(str, Enum):
    PENDING = "pending"
    ACCEPTED = "accepted"
    REJECTED = "rejected"

# 1. 新增：子资源表述，专门用于在绑定关系中展示用户信息
class BindingUser(BaseModel):
    user_id: int
    phone: str
    nickname: Optional[str] = None
    avatar_url: Optional[str] = None
# 【新增】：告诉 Pydantic 允许从 SQLAlchemy 的 User 对象读取属性
    model_config = ConfigDict(from_attributes=True)
# 2. 读表述：使用嵌套结构，清晰反映实体关系
class BindingResponse(BaseModel):
    binding_id: int
    status: BindingStatus
    created_at: datetime
    approved_at: Optional[datetime] = None
    # 嵌套结构，拒绝平铺数据库字段
    elderly: Optional[BindingUser] = None
    family: Optional[BindingUser] = None
    model_config = ConfigDict(from_attributes=True)
# class BindingCreate(BaseModel):
#     elderly_phone: str
#     family_id: Optional[int] = None

# 3. 写表述：发起绑定只需要老人手机号。family_id 绝对不能由前端传入
class BindingCreate(BaseModel):
    elderly_phone: str = Field(..., description="要绑定的老人手机号")

# 4. 写表述：状态更新（替代原来的 /approve 和 /reject）
class BindingStatusUpdate(BaseModel):
    status: BindingStatus = Field(..., description="目标状态：accepted 或 rejected")
# class BindingResponse(BaseModel):
#     binding_id: int
#     elderly_id: int
#     family_id: int
#     elderly_nickname: Optional[str] = None
#     elderly_phone: Optional[str] = None
#     family_nickname: Optional[str] = None
#     family_phone: Optional[str] = None
#     status: BindingStatus
#     created_at: datetime
#     approved_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class BindingUnbind(BaseModel):
    elderly_phone: str
    family_id: Optional[int] = None

